docs/lang/articles/misc/profiler.md

```diff
@@ -4,78 +4,125 @@ sidebar_position: 4
 
-```python {3,13}
+1. This profiler is automatically on.
+2. Call `ti.print_profile_info()` to display results in a hierarchical format.
+
+For example:
+
+```python
 import taichi as ti
 
-ti.init(ti.cpu, kernel_profiler=True)
+ti.init(arch=ti.cpu)
 var = ti.field(ti.f32, shape=1)
 
-
 @ti.kernel
 def compute():
     var[0] = 1.0
-
+    print("Setting var[0] =", var[0])
 
 compute()
-ti.print_kernel_profile_info()
-```
-
-The outputs would be:
-
-```
-[ 22.73%] jit_evaluator_0_kernel_0_serial             min   0.001 ms   avg   0.001 ms   max   0.001 ms   total   0.000 s [      1x]
-[  0.00%] jit_evaluator_1_kernel_1_serial             min   0.000 ms   avg   0.000 ms   max   0.000 ms   total   0.000 s [      1x]
-[ 77.27%] compute_c4_0_kernel_2_serial                min   0.004 ms   avg   0.004 ms   max   0.004 ms   total   0.000 s [      1x]
+ti.print_profile_info()
 ```
 
 :::note
-Currently the result of `KernelProfiler` could be incorrect on OpenGL
-backend due to its lack of support for `ti.sync()`.
+`ScopedProfiler` is a C++ class in the core of Taichi. It is not exposed to Python users.
 :::
 
-## ScopedProfiler
 
-1.  `ScopedProfiler` measures time spent on the **host tasks**
-    hierarchically.
-2.  This profiler is automatically on. To show its results, call
-    `ti.print_profile_info()`. For example:
+## KernelProfiler
 
-```python
-import taichi as ti
+`KernelProfiler` acquires the kernel profiling records from the backend, counts them in Python-scope, and prints the results to the console.
 
-ti.init(arch=ti.cpu)
-var = ti.field(ti.f32, shape=1)
+1. To enable this profiler, set `kernel_profiler=True` in `ti.init`.
+2. To display the profiling results, call `ti.print_kernel_profile_info()`. There are two modes of printing:
+    - In `'count'` mode (default), records with the same kernel name are counted as a profiling result,
+    and then presented in a statistical perspective.
+    - In `'trace'` mode, the profiler shows you a list of kernels that were launched on hardware during the profiling period.
+    This mode provides more detailed performance information and runtime hardware metrics for each kernel.
+3. To clear records in this profiler, call `ti.clear_kernel_profile_info()`.
 
+For example:
 
-@ti.kernel
-def compute():
-    var[0] = 1.0
-    print("Setting var[0] =", var[0])
+```python {3,13}
+import taichi as ti
 
+ti.init(ti.cpu, kernel_profiler=True)
+x = ti.field(ti.f32, shape=1024*1024)
 
-compute()
-ti.print_profile_info()
+@ti.kernel
+def fill():
+    for i in x:
+        x[i] = i
+
+for i in range(8):
+    fill()
+ti.print_kernel_profile_info('trace')
+ti.clear_kernel_profile_info() # clear all records
+
+for i in range(100):
+    fill()
+ti.print_kernel_profile_info() # default mode: 'count'
 ```
 
-`ti.print_profile_info()` prints profiling results in a hierarchical format.
+The outputs would be:
+
+```
+=========================================================================
+X64 Profiler(trace)
```

Comment:
i just notice this. `X64 Profiler` seems quite generic... What we have is actually a `Taichi Profiler running on X64 architecture`, or?

Reply:
fixed in #3049 , thanks!